src/Common/src/Interop/Unix/libgss/Interop.GssApiException.cs

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Microsoft.Win32.SafeHandles;
+
+internal static partial class Interop
+{
+    internal static partial class libgssapi
+    {
+        internal sealed class GssApiException : Exception
+        {
+            private Status _minorStatus;
+
+            public Status MinorStatus
+            {
+                get { return _minorStatus;  }
+            }
+
+            public GssApiException(string message) : base(message)
+            {
+            }
+
+            public GssApiException(Status majorStatus, Status minorStatus)
+                : this(SR.Format(SR.net_gssapi_operation_failed, majorStatus, minorStatus), majorStatus, minorStatus)
+            {
+            }
+
+            public GssApiException(string message, Status majorStatus, Status minorStatus)
+                : this(message)
+            {
+                HResult = (int)majorStatus;
+                _minorStatus = minorStatus;
+            }
+
+
+            public static GssApiException Create(string message)
```

Comment:
The `CreateSslException` methods do work before calling a constructor; and they're declared on the interop type. These Create methods just defer to the public constructor on the same type they're in.

If the Create methods were on the interop type, I could understand an information-hiding principle.  If there were being enough work done that it's hard to do in a base-ctor call, I could understand the factory method over a public constructor.  But since none of those reasons apply, I just don't understand what purpose they're serving.